server/api/migrations/versions/67b0169ba6b4_add_alert_history.py

```diff
@@ -0,0 +1,107 @@
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""add alert history
+
+Revision ID: 67b0169ba6b4
+Revises: fcf2ea01f99a
+Create Date: 2024-10-30 12:01:35.944309
+
+"""
+
+import os
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import mysql
+
+# revision identifiers, used by Alembic.
+revision = "67b0169ba6b4"
+down_revision = "fcf2ea01f99a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "alert_history",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column(
+            "activation_time", mysql.DATETIME(timezone=True, fsp=3), nullable=False
+        ),
+        sa.Column(
+            "name", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "project", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("data", sa.JSON(), nullable=True),
+        sa.Column(
+            "entity_id", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "entity_kind",
+            sa.String(length=255, collation="utf8mb3_bin"),
+            nullable=False,
+        ),
+        sa.Column(
+            "event_kind", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "severity", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("number_of_events", sa.Integer(), nullable=False),
+        sa.PrimaryKeyConstraint("activation_time", "id", name="_alert_history_uc"),
+    )
+    op.create_index(
+        "ix_alert_history_entity_id", "alert_history", ["entity_id"], unique=False
+    )
+    op.create_index(
+        "ix_alert_history_entity_kind", "alert_history", ["entity_kind"], unique=False
+    )
+    op.create_index(
+        "ix_alert_history_event_kind", "alert_history", ["event_kind"], unique=False
+    )
+    op.create_index("ix_alert_history_name", "alert_history", ["name"], unique=False)
+    op.create_index(
+        "ix_alert_history_project", "alert_history", ["project"], unique=False
+    )
+    op.create_index(
+        "ix_alert_history_severity", "alert_history", ["severity"], unique=False
+    )
+    # ### end Alembic commands ###
+
+    # Add initial partitioning
+    partition_interval = os.getenv("PARTITION_INTERVAL", "YEARWEEK")
+    default_partition_value = os.getenv("DEFAULT_PARTITION_VALUE", "2023-01-01")
```

Comment:
This migration will run once, and then the revision will be set to the revision defined in this file.
How will we be able to modify these partitioning values later?

Reply:
@TomerShor we don't have such a requirement. Technically it is possible, but according to the requirement we should not support this.

Reply:
So what if QA will want to have partitioning < 1 week?
Will it not be possible?

Reply:
@TomerShor it is possible by setting `PARTITION_INTERVAL` envvar on initial setup.

Reply:
From requirement:

```
partitioning granularity is configurable. This should be considered under the following restrictions:

* This configuration will not be dynamic nor will it be able to change during upgrades. Meaning that a system installed with partitioning granularity X will have to keep that granularity across future upgrades. There will not be any migration process.

* Given this limitation, this configuration (if available) will only be used by QA to facilitate testing. We should not document nor use it in customer deployments.
```